### ISSUE TITLE
Binary buffering?

### ISSUE THREAD

**Issue by Shadowstep33** [WebReflection/circular-json#22]
Hi, I love your library so far, great work!

I currently plan on utilizing it in a new project I am working on, but to really get it's full potential I'd love to be able to buffer my circular json as binary data (so I can reconstruct it piece-meal on the client, but still use it while it is being buffered). 

Is there a way I can accomplish this in conjunction with your library? From what I can tell so far, CircularJSON.stringify-ing my json before buffering it doesnt actually give me any advantages so far (other than the capability to even transmit the data in the first place ;] ) : I have to wait for the whole buffer to finish on the client before being able to use the data. I also can't target the more "relevant" data. 

Do you have any suggestions?

**Comment by WebReflection** [comment]
you need a way to buffer JSON in the first place, then I'd say it's quite straight forward because all revival operations are made when the object is already known.

```js
// plain Array
[o1, o2, o1, o3, o2]
// after CircularJSON
[o1,o2,~0,o3,~1]
```

while deserializing indexes (or keys) like `~0` or `~1` will always have a previously unserialized object as reference.

TL;DR this should work out of the box but only if you have a mechanism to stream JSON and parse it as stream
